.github/workflows/release.yml

```diff
@@ -1,158 +1,206 @@
 
   release:
     permissions:
       contents: write
     needs:
+      - pre-release
       - build-and-sign
-      - parse
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: ["ubuntu-latest"]
+        include: 
+          - ${{ needs.pre-release.outputs }}
+    runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v3
+      - name: Matrix description
+        run: |
+          echo "Running job with dry_run: '${{ inputs.dry_run }}', crate: '${{ matrix.crate }}', version: '${{ matrix.version }}', runtime: '${{ matrix.runtime }}', and is_shim: '${{ matrix.is_shim }}'."
+      - uses: actions/checkout@v4
       - name: Setup build env
         run: ./scripts/setup-linux.sh
       - name: Download artifacts
-        if: ${{ needs.parse.outputs.runtime != 'wasm' && needs.parse.outputs.runtime != 'wasm-test-modules' && needs.parse.outputs.runtime != 'oci-tar-builder'}}
+        if: ${{ matrix.is_shim == 'true' }}
         uses: actions/download-artifact@master
         with:
           path: release
+      - name: Tag the the release
+        if: ${{ !inputs.dry_run }}
+        run: |
+          git tag "${{matrix.crate}}/v${{matrix.version}}"
+          git push origin "${{matrix.crate}}/v${{matrix.version}}"
       - name: Create release
+        if: ${{ !inputs.dry_run }}
         run: |
-          gh release create ${{ github.ref }} --generate-notes --prerelease
+          gh release create 'refs/tags/${{matrix.crate}}/v${{matrix.version}}' --generate-notes --prerelease
         env:
           GH_TOKEN: ${{ github.token }}
-          RELEASE_NAME: ${{ needs.parse.outputs.crate }}/${{ needs.parse.outputs.version }}
+          RELEASE_NAME: ${{ matrix.crate }}/${{ matrix.version }}
       - name: Upload release artifacts
-        if: ${{ needs.parse.outputs.runtime != 'wasm' && needs.parse.outputs.runtime != 'wasm-test-modules' && needs.parse.outputs.runtime != 'oci-tar-builder'}}
+        if: ${{ matrix.is_shim == 'true' && !inputs.dry_run }}
         run: |
           for i in release/*/*; do
             gh release upload ${RELEASE_NAME} $i
           done
         env:
           GH_TOKEN: ${{ github.token }}
-          RELEASE_NAME: ${{ needs.parse.outputs.crate }}/${{ needs.parse.outputs.version }}
+          RELEASE_NAME: ${{ matrix.crate }}/${{ matrix.version }}
       - name: Cargo publish
-        run: cargo publish --package ${{ needs.parse.outputs.crate }} --verbose --locked
-        env:
-          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_PUBLISH_TOKEN }}
-      - name: Check crates.io ownership
-        run: |
-          cargo owner --add github:containerd:runwasi-committers ${{ needs.parse.outputs.crate }}
-          cargo owner --list ${{ needs.parse.outputs.crate }} | grep github:containerd:runwasi-committers
+        if: ${{ matrix.is_shim  != 'true' }}
+        run: cargo publish ${{ inputs.dry_run && '--dry-run' || '' }} --package ${{ matrix.crate }} --verbose --locked
```

Comment:
I think the step order here should be:

1. publish
2. tag repo
3. create release

if publishing fails we shouldn't be creating a release.  Also we can then add something to the release notes of the crate like 

```
Add the crate to you project:

cargo add containerd-shim-wasm@0.5.0
```